flow/connectors/mysql/qrep.go

```diff
@@ -115,30 +115,29 @@ func (c *MySqlConnector) GetQRepPartitions(
 	}
 
 	// Calculate the number of partitions
-	numPartitions := totalRows / numRowsPerPartition
-	if totalRows%numRowsPerPartition != 0 {
-		numPartitions++
-	}
+	adjustedPartitions := shared.AdjustNumPartitions(totalRows, numRowsPerPartition)
```

Comment:
ooc, is the main difference here limiting `maxPartitions` to 1000?

Reply:
yes, and adjusting numRowsPerPartition accordingly